### ISSUE TITLE
Remove / fix couch2pg \u0000 escaping

### ISSUE THREAD

**Issue by SCdF** [medic/couch2pg#6]
Postgres doesn't support storing JSON with \u0000 in strings.

We have some dodgy escaping code in couch2pg, but there is a good chance it's wrong. Specifically, it probably removes escaped backslashes more aggressively than it should.

We should work out how to solve this problem properly, instead of just naively regex escaping.

Or we should watch pg and once they support \u0000 properly remove the dodgy code.

**Comment by alxndrsn** [comment]
Postgres's built-in `escape()` function might be what we're looking for: https://www.postgresql.org/docs/9.3/functions-string.html#FUNCTIONS-STRING-OTHER

**Comment by SCdF** [comment]
Unfortunately I couldn't get the encode function working. I don't think it likes dealing with unicode string declarations:

```
scdf=# select encode('abcd', 'escape') as foo;
 foo
------
 abcd
(1 row)

scdf=# select encode('ab\u0000cd', 'escape') as foo;
ERROR:  invalid input syntax for type bytea
LINE 1: select encode('ab\u0000cd', 'escape') as foo;
```

However, I found "the real bug", which is:
a) we weren't escaping identifiers when we queried to delete documents
b) we were only string declared unicode, e.g. the 6 character '\u0000' sequence. We weren't escaping actual unicode, e.g. the one character 0x00 value. This latter version is what we get in the document's id from CouchDB
